### ISSUE TITLE
Check Jacobian gradient before correcting higher derivatives

### ISSUE THREAD

**Issue by simonsticko** [dealii/dealii#9682]
Correcting the 2nd and 3rd derivatives of the shape functions with the derivatives of the Jacobian is expensive. #9530 adds a function
 `higher_derivatives_need_correcting` in `fe/fe_poly.templates.h` 
that returns whether the correction terms need to be computed. Currently this function only checks if the mapping is Cartesian, but a more general condition would be to check the value of the `jacobian_grads_pushed_forward` at the quadrature points. Precisely what condition should be checked is open for discussion.

**Comment by bangerth** [comment]
The condition to check is whether the norm of `jacobian_grads_pushed_forward` is `<<` than the norm of `jacobian` divided by some length scale (e.g., the diameter of the cell we're on).

**Comment by masterleinad** [comment]
I don't think this is crucial for the release and would move the milestone unless there are volunteers to tackle it.

**Comment by peterrum** [comment]
@simonsticko How do we want to continue here?

**Comment by kronbichler** [comment]
I guess this is not critical, so we can postpone the target milestone (and if we happen to have time for this one we can still work on it).

**Comment by kronbichler** [comment]
I think we have substantially reduced costs in `MappingQ` related functions by #15035 and #15125 and I consider this issue addressed. Of course it would be nice to get further progress also for other mappings, but I suggest to remove it from the current target milestone, given how long this has been around unaddressed.
